libs/framework/include/celix_bundle_context.h

```diff
@@ -771,6 +771,12 @@ typedef struct celix_service_use_options {
      * and the bundle owning the service will also be provided to the callback.
      */
     void (*useWithOwner)(void *handle, void *svc, const celix_properties_t *props, const celix_bundle_t *svcOwner) OPTS_INIT;
+    /**
+     * @brief Call the provided callbacks from the caller thread directly if set, otherwise the callbacks will be called from the Celix event loop (most likely indirectly).
+     * Note that using blocking service in the Celix event loop is generally a bad idea, which should be avoided if possible.
+     */
+#define CELIX_SERVICE_USE_DIRECT          (1)
```

Comment:
I agree that `CELIX_SERVICE_USE_DIRECT` ideally should be the default (mail in dev mailing list)

Although it is good that "service on demand" can be triggered with useService(s), this is a more corner use case and especially useful for testing purpose.

Reply:
Yes. I add `CELIX_SERVICE_USE_SOD` to support "service on demand" if required. The user only has to set the flag explicitly, together with `CELIX_SERVICE_USE_DIRECT`.